ocrd_models/ocrd_page_user_methods.py

```diff
@@ -117,9 +117,9 @@ def get_recursive_reading_order(rogroup):
             reading_order = get_recursive_reading_order(reading_order)
         if reading_order:
             ret = dict([(region.id, region) for region in ret])
-            return [ret[region_id] for region_id in reading_order if region_id in ret]
-        else:
-            return ret
+            ret = [ret[region_id] for region_id in reading_order if region_id in ret]
+        ret = [r in ret if r.__class__.__name__.replace('RegionType', '') in regions
```

Comment:
There's a problem with that line (missing closing bracket, no `for`).

Reply:
thx i am aware but had to leave the office now, will finish pr later from home, hence the "wip".